questionnaire/questionnaire-dal.js

```diff
@@ -177,6 +177,34 @@ function questionnaireDAL(spec) {
         return result;
     }
 
+    async function getQuestionnaireMetadata(query) {
+        let result;
+        try {
+            if (query?.filter?.userId) {
+                result = await db.query(
+                    `SELECT id, questionnaire -> 'meta' -> 'questionnaireDocumentVersion' AS "questionnaire-version", created, modified, submission_status, questionnaire -> 'answers' -> 'user' -> 'user-id' AS "user-id" FROM questionnaire WHERE questionnaire -> 'answers' -> 'user' ->> 'user-id' = $1`,
+                    [query.filter.userId]
+                );
+                if (result.rowCount === 0) {
+                    throw new VError(
+                        {
+                            name: 'ResourceNotFound'
+                        },
+                        `User "${query.filter.userId}" not found`
+                    );
+                }
+            } else {
+                result = await db.query(
+                    "SELECT id, questionnaire -> 'meta' -> 'questionnaireDocumentVersion' AS \"questionnaire-version\", created, modified, submission_status, questionnaire -> 'answers' -> 'user' -> 'user-id' AS \"user-id\" FROM questionnaire"
```

Comment:
should this else block use the same pattern as the preceding if block above it?: get result, throw if rowCount is 0. Then return at the bottom of the function if it falls through to it

Reply:
Well the thinking was to get to this `else` branch the user has asked for all metadata. The result is either some metadata or no metadata. They've not provided any parameters that would generate an error.